### PR TITLE
Moving login to use keycloak login page

### PR DESCRIPTION
**Pull request by Vismayak**

### CODE REVIEW

**Comment by Rashmil-1999** [comment]
This PR is ready to be reviewed. To test it, there are 3 options: I suggest use tst or dev
1. Test locally, for this you need to have docker installed. Go to keycloak setup folder and run `docker-compose up -d` make sure port 8080 is free. Then in `asset.config.js` file set URL to `http://localhost:8080/auth/` under `keycloakConfig`. Then you can run `npm run start`. For the first time, you might have to go to keycloak on port 8080 to set the theme and redirect URLs in `react_auth` client.
2. Do nothing, just run `npm run start:tst` to connect to `tst` cluster. Note that there is no data and some components would not work fine. but the Login and logout should.
3. Run the usual `npm run start:dev` to connect to `dev` cluster. Everything should be as is on `https://incore-dev.ncsa.illinois.edu/` but the login page should have changed. Note that YWK is working on getting the theme on dev so you might see the old green theme. Just Ctrl/CMD + Shift + r to hard reload to pull new themes when they get added.

I suggest try option 2 and 3 ....1 only if you want to understand exactly what did we change on keycloak settings page.

**Review by ywkim312 — COMMENTED** [review]
Rashimil, let's keep the Dockerfile. This will be needed when we need to update the theme

**Comment by Rashmil-1999** [comment]
Ohh okay...

File: src/actions/index.js [hunk 1]
```diff
@@ -529,14 +529,31 @@ export const loginHelper = async (username, password) => {
...
+// 				type: SET_USER,
+// 				Authorization: `bearer ${json["access_token"]}`
+// 			});
+// 		} else {
+// 			return dispatch({
+// 				type: LOGIN_ERROR
+// 			});
+// 		}
+// 	};
+// }
```
Comment (longshuicy) [inline src/actions/index.js:547]: If not needed just remove. This logic will be in git anyway.

File: src/app.config.js [hunk 2]
```diff
@@ -43,4 +43,12 @@ config["slackInvitationLink"] = "https://bit.ly/in-core";
 config["slackWorkspaceLink"] = "https://in-core.slack.com";
 config["githubRelease"] = "https://github.com/IN-CORE/IN-CORE/releases/tag/";
 
-export default config;
+// Keycloak configuration
+config["keycloakConfig"] = {
+	// url: "http://localhost:8080/auth/", // for testing locally
+	url: `${hostname}/auth/`,
+	realm: "In-core",
+	clientId: "react-auth"
```
Comment (longshuicy) [inline src/app.config.js:51]: We have a config variable for that already. 
See line 7: `config["client_id"] = "react-auth";`

File: src/components/App.jsx [hunk 3]
```diff
@@ -179,12 +180,16 @@ class App extends Component {
...
-		});
+	handleKeycloakLogout() {
+		const redirectUri = `${location.origin}/`
+		try {
+			this.props.logout();
+			keycloak.logout({
+				redirectUri: redirectUri
+			});
+		} catch (error) {
+			console.log("Logout error", error);
```
Comment (longshuicy) [inline src/components/App.jsx:191]: Recently I start to realize that there is `console.error ` might be better to document error

File: src/components/App.jsx [hunk 4]
```diff
@@ -238,7 +243,7 @@ class App extends Component {
 		let group;
 
 		let contents = (
-			<Button color="inherit" href={"login"} className={classes.smallButton}>
+			<Button color="inherit" onClick={() => browserHistory.push("/Login")} className={classes.smallButton}>
```
Comment (longshuicy) [inline src/components/App.jsx:246]: Is it a typo or we change this route to be capitalized "Login"?

File: src/components/Login.jsx [hunk 5]
```diff
@@ -1,188 +1,57 @@
...
-							<Link href={config.signUpURL} className={classes.signUp} target="_blank">
-								Don't have an account? Sign up.
-							</Link>
-						</ImageListItem>
-					</ImageList>
-				</Paper>
-				<Version />
-			</div>
+		<div style={styles.container}>
+			<CircularProgress />
```
Comment (longshuicy) [inline src/components/Login.jsx:54]: The circular progress seems very convenient. Today I learned. :-)

**Review by longshuicy — CHANGES_REQUESTED** [review]
One additional comment: Old logos are not needed anymore. Maybe just remove them to be clean.

File: src/components/Login.jsx (hunk 5 above)
Comment (Rashmil-1999) [reply src/components/Login.jsx:54]: :D

File: src/components/App.jsx (hunk 4 above)
Comment (Rashmil-1999) [reply src/components/App.jsx:246]: I do not recall whether it was this PR or from earlier...

Comment (Rashmil-1999) [reply src/components/App.jsx:246]: It was there from before for all the components

File: CHANGELOG.md [hunk 6]
```diff
@@ -16,6 +21,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Changed
 - Remodel the landing page [#170](https://github.com/IN-CORE/incore-ui/issues/170)
 - Hazard viewer previewing hazard datasets [#172](https://github.com/IN-CORE/incore-ui/pull/172)
+- Using keycloak's login page instead of custom login page [#174](https://github.com/IN-CORE/incore-ui/issues/174)
```
Comment (longshuicy) [inline CHANGELOG.md:24]: do you want to remove that entry since it belongs to the Unreleased section

File: src/app.config.js [hunk 7]
```diff
@@ -43,4 +43,12 @@ config["slackInvitationLink"] = "https://bit.ly/in-core";
 config["slackWorkspaceLink"] = "https://in-core.slack.com";
 config["githubRelease"] = "https://github.com/IN-CORE/IN-CORE/releases/tag/";
 
-export default config;
+// Keycloak configuration
+config["keycloakConfig"] = {
+	// url: "http://localhost:8080/auth/", // for testing locally
+	url: `${hostname}/auth/`,
+	realm: "In-core",
```
Comment (longshuicy) [inline src/app.config.js:50]: could you replace the relam with config["client_id"] so we don't need to hardcde the info twice?

**Review by longshuicy — APPROVED** [review]

File: keycloakSetup/docker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+version: '3.7'
+
```
Comment (ywkim312) [inline keycloakSetup/docker-compose.yaml:2]: Maybe we want to keep this docker-compose file in the box folder and remove it from here?

File: src/app.config.js (hunk 7 above)
Comment (Rashmil-1999) [reply src/app.config.js:50]: I don't think there is a variable for `In-core` I replaced it for client id as it was there...

File: CHANGELOG.md (hunk 6 above)
Comment (Rashmil-1999) [reply CHANGELOG.md:24]: Ahh shit!  Good catch.. I have removed it from the released section...

File: keycloakSetup/docker-compose.yaml (hunk 8 above)
Comment (Rashmil-1999) [reply keycloakSetup/docker-compose.yaml]: Okay... done...

**Review by ywkim312 — APPROVED** [review]

File: src/app.config.js (hunk 7 above)
Comment (longshuicy) [reply src/app.config.js:50]: Ah right my bad I confused it with the "react-auth" client id

**Comment by ywkim312** [comment]
@Rashmil-1999 I am very sorry to request that would you be able to update this PR by merging with the latest develop? So we can deploy this PR with google anlaytics in dev cluster?

**Comment by Rashmil-1999** [comment]
yes, let me do that...

**Pull request merged**